ranges.md

```diff
@@ -0,0 +1,384 @@
+
+1. The terms SemVer and SemVer Version refer to version strings conforming
+   with the [Semantic Versioning specification
+   2.0.0](https://semver.org/spec/v2.0.0.html)
+
+1. A SemVer Range is a set of 1 or more `Comparator Sets`, joined by `||`.
+   A SemVer Version is included by the SemVer Range if it is included by
+   one or more of the Comparator Sets.
+
+1. A Comparator Set is a set of 1 or more `Comparators`, joined by 1 or
+   more space characters.  (That is, `' '`, ASCII value 32.)
+
+1. A Comparator is an `Operator` and a `Version`.  The Comparator Version
+   is a SemVer string.  The Operator is one of: `>`, `>=`, `<`, `<=`, `=`.
+   The Operator MAY be omitted, in which case `=` is assumed.  For example,
+   `>1.2.3`, `<=2.1.3`, `5.6.2`.
+
+    1. Comparators with the Operator of `=` will include a given SemVer
+       Version if their Version has precedence equal to that of the SemVer
+       Version being considered.
+
+    1. Comparators with the Operator of `>` will include a given SemVer
+       Version if their Version has precedence greater than that of the
+       SemVer Version being considered.
+
+    1. Comparators with the Operator of `>=` will include a given SemVer
+       Version if their Version has precedence greater than or equal to
+       that of the SemVer Version being considered.
+
+    1. Comparators with the Operator of `<` will include a given SemVer
+       Version if their Version has precedence less than that of the
+       SemVer Version being considered.
+
+    1. Comparators with the Operator of `<=` will include a given SemVer
+       Version if their Version has precedence less than or equal to that
+       of the SemVer Version being considered.
+
+1. Inclusion of PRERELEASE idenfiers in Comparator Sets.
+
+    SemVer Versions that include PRERELEASE identifiers are assumed to be
+    unstable and/or incomplete.
+
+    Additionally, the presence of PRERELEASE identifiers in a version such
+    as `I.J.K-PRERELEASE` causes the SemVer Version to have a _lower_
+    precedence than `I.J.K`.  Thus, it could cause a Range such as `<2.0.0`
+    to include the version `2.0.0-rc.0`, which likely contains breaking API
+    changes that the consumer is not prepared to handle.
+
+    To avoid surprising behavior for consumers, the inclusion of SemVer
+    Versions with PRERELEASE identifiers is as follows.
+
+    1. SemVer Versions containing PRERELEASE identifiers MUST NOT be
+       included by a SemVer Comparator Set unless they are included by all
+       of the Comparators in the Comparator Set, and one or more of the
+       following conditions are met:
+
+        1. The implementation has provided the user with an option to
+           explicitly include PRERELEASE versions, and the user has set this
+           option.  This SHOULD NOT be the default behavior when resolving
+           software dependencies.
+
+        1. One or more of the Comparators in the Comparator Set have the
```

Comment:
this differs from cocoapods (i think, i might also be thinking of bundler) behavior whereby the major/minor/patch don't need to match, it's only required that at least one comparator have a version with any prerelease specifier in it. (e.g. `> 0.a` allows matching any pre-release)